docs/advanced/styling.md

```diff
@@ -8,20 +8,321 @@ has_toc: false
+
+- `task-due-today` (or `task-start-today`, `task-done-today` etc) represents today.
+- `task-due-future-1d` (or `task-start-future-1d`) represents 1 day in the future, i.e. tomorrow.
+- `task-due-past-1d` (or `task-start-past-1d`) represents 1 day in the past, i.e. yesterday.
+- These specific classes are added up to 7 days in the future or past, e.g. `task-scheduled-future-7d` or `task-due-past-7d`.
+- Dates that are further than 7 days in the future or past are given a `far` postfix, e.g. `task-scheduled-future-far` or `task-due-past-far`.
+
+Specific classes are added to both their corresponding components (e.g. to the due date component) and also to the complete task `li`, to make it easy for a CSS rule to style a complete task according to some property (e.g. color differently the complete task if it's due today) or just one relevant component.
+
+**Tip:** [CSS wildcard selectors](https://www.geeksforgeeks.org/wildcard-selectors-and-in-css-for-classes/) are a good way to select all past dates or future dates at once -- just use `[class*="past-"] .task-due ...` to address all overdue tasks, for example.
+
+## More Classes
+
+The following additional components have the following classes:
 
 | Class                          | Usage                                                                                                           |
 | ------------------------------ | --------------------------------------------------------------------------------------------------------------- |
-| plugin-tasks-query-result      | This is applied to the UL used to hold all the tasks, each task is stored in a LI.                              |
 | plugin-tasks-query-explanation | This is applied to the PRE showing the query's explanation when the `explain` instruction is used.              |
-| plugin-tasks-list-item         | This is applied to the LI that holds each task and the INPUT element for it.                                    |
 | tasks-backlink                 | This is applied to the SPAN that wraps the backlink if displayed on the task.                                   |
 | tasks-edit                     | This is applied to the SPAN that wraps the edit button/icon shown next to the task that opens the task edit UI. |
 | tasks-urgency                  | This is applied to the SPAN that wraps the urgency score if displayed on the task.                              |
-| task-list-item-checkbox        | This is applied to the INPUT element for the task.                                                              |
 | tasks-group-heading            | This is applied to H4, H5 and H6 group headings                                                                 |
 
-{: .released }
-`tasks-group-heading` was introduced in Tasks 1.6.0.<br>
-`plugin-tasks-query-explanation` was introduced in Tasks 1.19.0.
+> `tasks-group-heading` was introduced in Tasks 1.6.0.<br>
+> `plugin-tasks-query-explanation` was introduced in Tasks 1.19.0.
+
```

Comment:
FYI I recently changed the way that 'feature x was introduced in [version number]' is formatted.

It's documented here:
https://github.com/obsidian-tasks-group/obsidian-tasks/blob/main/CONTRIBUTING.md#version-numbers-in-documentation

The new callout title got lost in this change, so please could it be reverted?
Also, it occurs to me that this would be the perfect place to add 3 more lines, for the 3 classes that were removed... for example:

```text
- `task-list-item-checkbox` was removed in Tasks X.Y.Z.
```

Which reminds me further... Virtually this whole page will need the standard message to say which version these new features were released in.

Reply:
I fixed the callout title, but for the whole page I'm unsure what to do, because I have no way of knowing which version of Tasks this will eventually be merged to. Would you like to add a callout for a specific version?

Reply:
Ah right. If you have time, please put in literally X.Y.Z.

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/main/CONTRIBUTING.md#version-numbers-in-documentation

Reply:
Sure, added.